\documentclass{bioinfo}
\usepackage{fancyvrb}
\copyrightyear{2005}
\pubyear{2005}

\begin{document}
\firstpage{1}

\title[short Title]{Brain: Biomedical Knowledge Manipulation}
\author[Croset S \textit{et~al}]{Samuel Croset\,$^{1,\footnote{to whom correspondence should be addressed}}$,
John Overington\,$^{1}$ and Dietrich Rebholz-Schuhmann\,$^{1}$}
\address{$^{1}$EMBL European Bioinformatics Institute, Wellcome Trust Genome Campus, Hinxton, Cambridge, CB10 1SD UK}

\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}

\editor{Associate Editor: XXXXXXX}

\maketitle

\begin{abstract}

\section{Summary:}
Brain is a software library facilitating the creation and manipulation of ontologies and knowledge-bases represented in the
Web Ontology Language (OWL).

\section{Availability and Implementation:}
The Java source code and the library are freely available at https://github.com/loopasam/Brain
and on the Maven Central repository (GroupId: uk.ac.ebi.brain).
The documentation is available at https://github.com/loopasam/Brain/wiki.

\section{Contact:}
\href{croset@ebi.ac.uk}{croset@ebi.ac.uk}

\section{Supplementary information:}
Supported features and a comparative table are available at the journal's web site.

\end{abstract}
\section{Motivation}
Knowledge bases, a concept from computer science (see \citealp{Krotzsch2012} for an introduction),
could be a solution to improve the interoperability and the value of the large amount of biomedical information available online.
At the time of writing, a few options are available to handle such knowledge bases: Complex
libraries as the OWL-API \citep{MatthewHorridge2011} or didactic graphical user interfaces such as
Protege \citep{StanfordCenterforBiomedicalInformaticsResearch} or TopBraid (ref). An intermediary framework,
OWLTools \citep{MungallC},
provides some methods to query biomedical knowledge bases, but the interaction with the library is mostly done via command-lines which
limits the scale of projects that can be build with it.
Brain - the software library presented in this manuscript addresses this matter and provides a comprehensive and simplified
interface, dedicated to the programmatic creation and query of biomedical knowledge bases. The library aims at bridging the gap between
graphical user interfaces and the OWL-API and is particularly usefull to develop web applications.
Brain has a particular focus on the EL profile of OWL, as it covers the majority of biomedical
use-cases and enables good performances and scalability. 

\section{Scalable knowledge bases}
The Web Ontology Language (OWL) derives from Description Logic and
has been designed to capture the knowledge of a domain of interest in the form of a structured vocabulary \citep{W3COWLWorkingGroup}.
This feature makes it particularly
interesting from the perspective of the life sciences, since a number of ontologies and classification schemes
have been developed from the origin of
the discipline. Brain focuses on a particular profile of OWL, called EL
which consists of a subset of the constructs available in the original language \citep{Motik2009}. This profile is
designed to be \emph{tractable}, meaning that the axioms available have a polynomial complexity
and are therefore easier to compute than the full version of OWL.
Brain primarily supports the OWL 2 EL profile for its computational properties and suitability for real-life biomedical applications,
where millions of axioms could be potentially extracted from complex repositories such as ChEMBL \citep{Gaulton2012}.
Moreover, the EL profile is expressive enough to cover a good portion of biomedical knowledge: Most of Open Biomedical Ontologies (OBO)
such as the Gene Ontology (GO - \citealp{Ashburner2000}) or
the Chemical Entities of Biological Interest (ChEBI - \citealp{DeMatos2010}) are already included in this
profile, opening doors to large-scale meaningful data integration.
Brain builds on the top of Elk, a fast reasoner dedicated to EL ontologies \citep{YevgenyKazakov2011}. Elk shows good
performances at handling
large datasets and offers the possibility to run some reasoning tasks in parallel; therefore
clusters or multicore architecture can scale the speed of reasoning as more data are added to the knowledge base.
Brain wraps and simplifies the interaction with Elk while still leaving the possibility to fine tune
the configuration for advanced users.

\section{Library features}
Brain is implemented as a facade, leveraging the access to the OWL-API and providing a series of convenience methods for common
use-cases encountered in the biomedical domain. In order to simplify the interaction with the OWL-API, Brain has been designed according to
a series of features described below.
\subsection{Unique ontology}
An instance of a Brain object hold a reference to only one knowledge base. It is yet possible to import some external ontologies, either
stored locally or via a network but Brain will always merge the added information to the existing knowledge base.
\subsection{Unique short form names}
The names (short forms) of OWL entities handled by a Brain object have to be unique. It is for example not possible to add
an OWL class http://www.example.org/Cell to the ontology if an OWL entity with short form ``Cell'' already exists.
Despite being in contradiction with some fundamental Semantic Web principles, this design prevent ambiguous queries and integration when 
short forms are used and hides as much as possible the cumbersome interaction with prefixes and Internationalized Resource Identifiers (IRI).
\subsection{Typeless interaction}
The interaction with the library relies on the user-friendly Manchester syntax entered as string \citep{Horridge2006}. 
This choice permits to move away from the cumbersome creation of Java objects and is particularly suitable in a web server environment where
requests are likely to be typeless. Using strings as input also speeds the production
and flexibility of the code written to move to an OWL representation from a relational or flat-file database for example.
\subsection{Error-handling}
Because the interaction with Brain is built around strings rather than Java objects, a special care has to be put on
exceptions handling in order to safely maintain the correct execution of the program. Brain throws different types of error tailored 
to the operation performed by the user. This feature is mandatory in large applications and helps to maintain the consistency of the underlying
knowledge base or to program a handling in case of problem.
\subsection{Knowledge integration}
An interesting feature brought by the Semantic Web and OWL is the possibility to merge information based
on the unique IRI of the entities described. The library supports the loading and integration of external knowledge bases as well as 
references to external entities. Data from different sources can therefore be easily connected and reason over by Brain.
\subsection{Querying}
Brain is oriented towards efficient querying of OWL knowledge bases.
Powerful questions can be formulated over the knowledge base using the Manchester syntax, abstracting
away complex interaction with the Java object provided by the OWL-API. An example of question answering over the GO using Brain
is compared against a traditional SQL query in the supplementary material.

\begin{figure}[h]
\begingroup
\fontsize{7pt}{8pt}\selectfont
\begin{Verbatim}[frame=single]
//You will always need a Brain object
Brain brain = new Brain();
//Add the OWL classes
brain.addClass("Nucleus");
brain.addClass("Cell");
//Add the OWL object property
brain.addObjectProperty("part-of");
//Declare the axiom
brain.subClassOf("Nucleus", "part-of some Cell");
//Query the knowledge-base for indirect subclasses of an expression
List<String> subClasses = brain.getSubClasses("part-of some Cell", false);
//Free the resources used by the reasoner
brain.sleep();
//Save the ontology
brain.save("your/path/to/ontology.owl");
\end{Verbatim}
\endgroup
\caption{Example of commented implementation of an axiom using Brain; axiom expressed in natural language:
 \textit{The nucleus is part of some cells}. Same axiom described in OWL using the Manchester syntax: 
 \textit{Nucleus subClassOf part-of some Cell.}}
\end{figure}

The full list of currently supported constructs is available in the supplementary material.


\section{Conclusion}
Brain is an open-source Java library designed to build and query biomedical knowledge bases or OWL ontologies.
The library is centered on the EL profile, and designed to be suitable and scalable for biomedical knowledge representation.
The convenience methods provided by Brain should simplify the development of biomedical knowledge-bases and allow developers
to increase their productivity while effectively dealing with data integration challenges.

\section*{Acknowledgement}
\paragraph{Funding\textcolon}
This work was supported by funding from the EMBL member states. Samuel Croset is a member of Darwin College, University of Cambridge.

% \bibliographystyle{natbib}
% \bibliographystyle{achemnat}
% \bibliographystyle{plainnat}
% \bibliographystyle{abbrv}
% \bibliographystyle{bioinformatics}
% 
% \bibliographystyle{plain}
% 
% \bibliography{document}

\begin{thebibliography}{}

\bibitem[Ashburner {\em et~al.}(2000)Ashburner, Ball, Blake, Botstein, Butler,
  Cherry, Davis, Dolinski, Dwight, Eppig, Harris, Hill, Issel-Tarver,
  Kasarskis, Lewis, Matese, Richardson, Ringwald, Rubin, and
  Sherlock]{Ashburner2000}
Ashburner, M., Ball, C.~A., Blake, J.~A., Botstein, D., Butler, H., Cherry,
  J.~M., Davis, A.~P., Dolinski, K., Dwight, S.~S., Eppig, J.~T., Harris,
  M.~A., Hill, D.~P., Issel-Tarver, L., Kasarskis, A., Lewis, S., Matese,
  J.~C., Richardson, J.~E., Ringwald, M., Rubin, G.~M., and Sherlock, G.
  (2000).
\newblock {Gene Ontology: tool for the unification of biology}.
\newblock {\em Nature Genetics\/}, {\bf 25}(1), 25--29.

\bibitem[Crosswell and Thornton(2012)Crosswell and Thornton]{Crosswell2012}
Crosswell, L.~C. and Thornton, J.~M. (2012).
\newblock {ELIXIR: a distributed infrastructure for European biological data.}
\newblock {\em Trends Biotechnol.}, {\bf 30}, 241----242.

\bibitem[{De Matos} {\em et~al.}(2010){De Matos}, Alc\'{a}ntara, Dekker, Ennis,
  Hastings, Haug, Spiteri, Turner, and Steinbeck]{DeMatos2010}
{De Matos}, P., Alc\'{a}ntara, R., Dekker, A., Ennis, M., Hastings, J., Haug,
  K., Spiteri, I., Turner, S., and Steinbeck, C. (2010).
\newblock {Chemical Entities of Biological Interest: an update}.
\newblock {\em Nucleic Acids Research\/}, {\bf 38}(Database issue), D249--D254.

\bibitem[Gaulton {\em et~al.}(2012)Gaulton, Bellis, Bento, Chambers, Davies,
  Hersey, Light, McGlinchey, Michalovich, Al-Lazikani, and
  Overington]{Gaulton2012}
Gaulton, A., Bellis, L.~J., Bento, A.~P., Chambers, J., Davies, M., Hersey, A.,
  Light, Y., McGlinchey, S., Michalovich, D., Al-Lazikani, B., and Overington,
  J.~P. (2012).
\newblock {ChEMBL: a large-scale bioactivity database for drug discovery}.
\newblock {\em Nucleic Acids Research\/}, {\bf 40}(Database issue), D1100--7.

\bibitem[{Horridge} and {Bechhofer}(2011){Horridge} and
  {Bechhofer}]{MatthewHorridge2011}
{Horridge} and {Bechhofer} (2011).
\newblock {The OWL API: A Java API for OWL Ontologies.}
\newblock {\em Semantic Web Journal\/}, pages 11--21.

\bibitem[Horridge {\em et~al.}(2006)Horridge, Drummond, Goodwin, Rector,
  Stevens, and Wang]{Horridge2006}
Horridge, M., Drummond, N., Goodwin, J., Rector, A., Stevens, R., and Wang,
  H.~H. (2006).
\newblock {The Manchester OWL Syntax}.
\newblock {\em Syntax\/}, {\bf 216}, 10--11.

\bibitem[{Kazakov} {\em et~al.}(2011){Kazakov}, {Markus Kr\"{o}tzsch}, and
  {Franti\v{s}ek Siman\v{c}\'{\i}k}]{YevgenyKazakov2011}
{Kazakov}, {Markus Kr\"{o}tzsch}, and {Franti\v{s}ek Siman\v{c}\'{\i}k} (2011).
\newblock {Concurrent Classification of EL Ontologies}.
\newblock {\em Proceedings of the 10th International Semantic Web Conference
  (ISWC'11)\/}, {\bf 7032}.

\bibitem[Kr\"{o}tzsch {\em et~al.}(2012)Kr\"{o}tzsch, Simanˇ, and
  Horrocks]{Krotzsch2012}
Kr\"{o}tzsch, M., Simanˇ, F., and Horrocks, I. (2012).
\newblock {A Description Logic Primer}.
\newblock {\em Language\/}, {\bf abs/1201.4}(January), 1--16.

\bibitem[{Manola} and {Miller}(2004){Manola} and {Miller}]{FrankManola}
{Manola} and {Miller} (2004).
\newblock {RDF Primer}.

\bibitem[Motik {\em et~al.}(2009)Motik, Grau, Horrocks, Wu, Fokoue, and
  Lutz]{Motik2009}
Motik, B., Grau, B.~C., Horrocks, I., Wu, Z., Fokoue, A., and Lutz, C. (2009).
\newblock {OWL 2 Web Ontology Language Profiles}.
\newblock {\em Language\/}, {\bf 2009}(October), 1--53.

\bibitem[Mungall(2012)Mungall]{MungallC}
Mungall (2012).
\newblock {OWLTools}.

\bibitem[{Stanford Center for Biomedical Informatics Research}(2012){Stanford
  Center for Biomedical Informatics
  Research}]{StanfordCenterforBiomedicalInformaticsResearch}
{Stanford Center for Biomedical Informatics Research} (2012).
\newblock {Prot\'{e}g\'{e} Project}.

\bibitem[{W3C OWL Working Group}(2009){W3C OWL Working
  Group}]{W3COWLWorkingGroup}
{W3C OWL Working Group} (2009).
\newblock {OWL 2 Web Ontology Language Document Overview}.

\end{thebibliography}

\end{document}
