\documentclass{bioinfo}
\copyrightyear{2005}
\pubyear{2005}

\begin{document}
\firstpage{1}

\title[short Title]{Brain: Biomedical Knowledge Manipulation}
\author[Sample \textit{et~al}]{Samuel Croset\,$^{1,\footnote{to whom correspondence should be addressed}}$, JPO\,$^{2}$ and DRS\,$^{2}$}
\address{$^{1}$Department of XXXXXXX, Address XXXX etc.\\
$^{2}$Department of XXXXXXXX, Address XXXX etc.}

\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}

\editor{Associate Editor: XXXXXXX}

\maketitle

\begin{abstract}

\section{Summary:}
This section should summarize the purpose/novel features of the program in one or two sentences.

\section{Availability and Implementation:}
This section should state software availability if the paper focuses mainly on 
software development or on the implementation of an algorithm. Examples are:
Freely available on the web at http://www.example.org. Website implemented in Perl, MySQL and 
Apache, with all major browsers supported'; or 'Source code and binaries freely available for download at URL, 
implemented in C++ and supported on linux and MS Windows'. The complete address (URL) should be given. If the manuscript 
describes new software tools or the implementation of novel algorithms the software must be freely available to non-commercial 
users. Authors must also ensure that the software is available for a full TWO YEARS following publication. The editors of 
Bioinformatics encourage authors to make their source code available and, if possible, to provide access through an open source 
license see www.opensource.org for examples.


\section{Contact:}
\href{croset@ebi.ac.uk}{croset@ebi.ac.uk}

\section{Supplementary information:}
Links to additional figures/data available on a web site, or 
reference to online-only Supplementary data available at the journal's web site.

\end{abstract}
Relational databases and ontologies hold most of the available structured biomedical information. The content of these repositories is often
extracted from scientific literature by manual curation with the help of text-mining tools. The transformation from raw text into
structured data is most important, as the curated information can then be classified, managed and queried more easily. Databases facilitate
the re-use of previous work in a computer-friendly manner and support the biomedical knowledge to scale-up. 
The availability of structured data enables the formulation of queries with the Structured Query Language (SQL) in order to retrieve the desired 
information over relational databases. For instance, it is possible to
quickly find the cellular location of a protein just by looking at its entry on Uniprot and without having to read several scientific articles
on the topic. The query language used and the structure of the database are the factors limiting the complexity of the questions that can be 
formulated over the biomedical information. For example, relational databases are powerful to store values about instance data such as the
sequence or the name of a gene product, but are daunting to use to represent hierarchical structures (ref) such as the Gene Ontology. Therefore
queries resolving around hierarchies are going to be complicated to formulate using SQL and will not necessarly deliver a complete answer. For
instance, simple queries such as retrieving the list of all the mammals from the NCBI taxonomy involves retrieving first the direct descendants
of the mammals and for each of them recursively retrieving their descendants. This type of construct is not trivial to implement
and is not yet supported natively by open-source solution like MySQL or PostgreSQL. 


//interoperability
//leverage
//



Ontologies and databases in biology are somehow used differently than in other fields. They essentially serve as read-only 
resources where the researchers come to browse or download the content. The majority of the time, no transactions or updates 
are hapenning as opposed to services such as Amazon or Ebay.

- The future of biomedical databases (ELIXIR) is leverage of their content via interoperability with other resources.
OWL provide the means to do this, databases don't.
- Features: Creation and storage of OWL knowledge-bases. 
Import of external knowledge-bases/ontologies. 
Simplification of interaction in regards to OWL-API. Fast classification time (Elk reasoner). 
Support complex queries via inference. Fast and suitable to be used in production.
- Evaluation: MySQL build of Go is compared versus OWL build of GO (identical content - different representation).
A series of biomedical questions will be answered in SQL and OWL respectively on the MySQL and on the OWL ontology.
Comparison of performances --> Brain is fast and scalable (thread friendly implementation).

Problems:
- Need to be able to formulate complex queries --> hidden/implicit knowledge - inference.
- Need for interoperability among resources --> semantic web technologies.
- Need to be able to leverage resources --> benefit from integration
- Relational databases are struggling with biomedical hierarchical representations. Instance versus classes
- Databases in biology are used as read-only resources
- Databases don't support inference natively but it could be overcome by a
 pre-processing step (similar to a classification for a knowledge base).

Solution:
- Semantic Web technologies could be slow --> OWL 2 EL.
- Advantages of triple -stores --> none, they still are relational flat data.
- Simplying interaction on OWL-API, tracktable problem. Focusing on biology, not computer science.
- More and more ontologies are represented in OWL.

OWL 2 EL building blocks:
- Individual, properties, classes
- Axioms
- Reasoner

- relation with description logic
- tracktability argument + definition
- OWL --> query and representation language
- table comparison query + times



\section*{Acknowledgement}
These should be included at the end of the text and not in footnotes. Please ensure you acknowledge all 
sources of funding, see funding section below.
Details of all funding sources for the work in question should be given 
in a separate section entitled 'Funding'. This should appear before the 'Acknowledgements' section.

\paragraph{Funding\textcolon}
The following rules should be followed:
The sentence should begin: This work was supported by …
 
The full official funding agency name should be given, i.e. National Institutes of Health, not NIH 
(full RIN-approved list of UK funding agencies) Grant numbers should be given in brackets as follows: grant number xxxx 
Multiple grant numbers should be separated by a comma as follows: grant numbers xxxx, yyyy 
Agencies should be separated by a semi-colon (plus ‘and’ before the last funding agency) 
Where individuals need to be specified for certain sources of funding the following text should be added after the
relevant agency or grant number to author initials.
Oxford Journals will deposit all NIH-funded articles in PubMed Central. See Depositing articles in repositories – 
information for authors for details. Authors must ensure that manuscripts are clearly indicated as NIH-funded using the guidelines above.

\bibliographystyle{natbib}
\bibliographystyle{achemnat}
\bibliographystyle{plainnat}
\bibliographystyle{abbrv}
\bibliographystyle{bioinformatics}

\bibliographystyle{plain}

\bibliography{document}

\end{document}
